tests/e2e_logging.py

```diff
@@ -17,6 +17,99 @@
 from loguru import logger as LOG
 
 
+def test_update_lua(network, args):
+    if args.package == "libluagenericenc":
+        LOG.info("Updating Lua application")
+        primary, _ = network.find_primary()
+
+        check = infra.ccf.Checker()
+
+        # Create a new lua application file (minimal app)
+        # TODO: Writing to file will not be required when memberclient is deprecated
+        new_app_file = "new_lua_app.lua"
+        with open(new_app_file, "w") as qfile:
+            qfile.write(
+                """
+                            return {
+                            ping = [[
+                                tables, args = ...
+                                return {result = "pong"}
+                            ]],
+                            }"""
+            )
+
+        network.set_lua_app(primary, new_app_file)
+        with primary.user_client(format="json") as c:
+            check(c.rpc("ping", params={}), result="pong")
+
+            LOG.debug("Check that former endpoints no longer exists")
+            for endpoint in [
+                "LOG_record",
+                "LOG_record_pub",
+                "LOG_get",
+                "LOG_get_pub",
+            ]:
+                check(
+                    c.rpc(endpoint, params={}),
+                    error=lambda e: e is not None
+                    and e["code"] == infra.jsonrpc.ErrorCode.METHOD_NOT_FOUND.value,
+                )
+    else:
+        LOG.warning("Skipping Lua app update as application is not Lua")
+
+    return network
+
+
+# Expects at least two nodes
+# Expects the libloggingenc application
```

Comment:
Can we express these requirements in code, rather than as comments? ie - add a `bool meets_requirements(network, args)` function, which checks the node count, and calls `listMethods` RPC to make sure `LOG_record`, `LOG_get` etc are supported? Then the randomiser test can dynamically select from the subset of tests which are currently valid.